src/Compilers/Core/Portable/Emit/EditAndContinueMethodDebugInformation.cs

```diff
@@ -348,12 +350,15 @@ internal void SerializeStateMachineStates(BlobBuilder writer)
             writer.WriteCompressedInteger(StateMachineStates.Length);
             if (StateMachineStates.Length > 0)
             {
+                // Negative syntax offsets are rare - only when the syntax node is in an initializer of a field or property.
+                // To optimize for size calculate the base offset and adds it to all syntax offsets. In common cases (no negative offsets)
+                // this base offset is gonna be 0. Otherwise it's gonna be the lowest negative offset.
```

Comment:
>this base offset is gonna be 0. Otherwise it's gonna be the [](http://example.com/codeflow?start=19&length=59)

Perhaps "... this base offset _will be_ 0. Otherwise, _it will be_ the ..."